pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1192,19 +1197,27 @@ func encodeSystemAdapterInfo(dpcl types.DevicePortConfigList) *info.SystemAdapte
 
 		dps.Ports = make([]*info.DevicePort, len(dpc.Ports))
 		for j, p := range dpc.Ports {
-			dps.Ports[j] = encodeNetworkPortConfig(&p)
+			dps.Ports[j] = encodeNetworkPortConfig(ctx, &p)
 		}
 		sainfo.Status[i] = dps
 	}
 	log.Debugf("encodeSystemAdapterInfo: %+v\n", sainfo)
 	return sainfo
 }
 
-func encodeNetworkPortConfig(npc *types.NetworkPortConfig) *info.DevicePort {
+func encodeNetworkPortConfig(ctx *zedagentContext,
+	npc *types.NetworkPortConfig) *info.DevicePort {
+	aa := ctx.assignableAdapters
+
+	ibPtr := aa.LookupIoBundlePhylabel(npc.Phylabel)
```

Comment:
Please put this line just about the nil check. Makes for more readable code.

Reply:
Good Point.. Done..